web/src/components/tasks/business-formation/business/BusinessStep.tsx

```diff
@@ -1,69 +1,111 @@
 import { MainBusiness } from "@/components/tasks/business-formation/business/MainBusiness";
+import { MainBusinessAddressNj } from "@/components/tasks/business-formation/business/MainBusinessAddressNj";
+import { MainBusinessForeignAddressFlow } from "@/components/tasks/business-formation/business/MainBusinessForeignAddressFlow";
 import { PartnershipRights } from "@/components/tasks/business-formation/business/PartnershipRights";
 import { Provisions } from "@/components/tasks/business-formation/business/Provisions";
 import { BusinessFormationTextBox } from "@/components/tasks/business-formation/BusinessFormationTextBox";
 import { BusinessFormationContext } from "@/contexts/businessFormationContext";
 import { useConfig } from "@/lib/data-hooks/useConfig";
-import { ReactElement, useContext } from "react";
+import { Fragment, ReactElement, useContext, useMemo } from "react";
 
 export const BusinessStep = (): ReactElement => {
   const { Config } = useConfig();
   const { state } = useContext(BusinessFormationContext);
 
+  const isForeign = useMemo(
+    () => state.formationFormData.businessLocationType !== "NJ",
+    [state.formationFormData.businessLocationType]
+  );
+
+  const isNonprofit = state.formationFormData.legalType === "nonprofit";
+
+  const addressSection = (): ReactElement =>
+    isForeign ? <MainBusinessForeignAddressFlow /> : <MainBusinessAddressNj />;
+
+  const lpSection = (): ReactElement | null => {
```

Comment:
[pebble] I think the pattern we've used elsewhere in the app is to return an empty element instead of `null`.

Reply:
This came up in discussion with @FarazA22 on my ticket as well, and wanted to jump in because I *do* think we return null all over the place in our app instead of empty element.

Everywhere in our code, we prefer to use `(shouldShowX && <Element />)` over `(shouldShowX ? <Element /> : <></>)`

When `shouldShowX` is `false` or `undefined`, nothing renders, by usage of the "JSX hole" which [the docs say are logically the same](https://react.dev/learn/conditional-rendering#logical-and-operator-) as returning `undefined` or `null` in your JSX tree.

So rewriting `shouldShowX && <Element />` into a function with a return type DOES make sense to have that function return ReactElement or null, imo. It's a closer mirror to the precise situation it was refactoring from.

In addition, I'd argue it's preferable. I don't think empty element is a good / useful thing to have in our code, which is why I prefer `&&` over the empty-element ternary above.

There is, of course, [a pretty good argument](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx) for ternaries over `&&` because of the danger with putting a `0` mistakenly on the left-hand-side. If we're concerned over that (validly), we should discuss as a team. Even if we do switch to ternary though, I think `(shouldShowX ? <Element /> : null)` is a more clear representation of what we want, and is preferable.